### ISSUE TITLE
msg="grpc: error dialing." err="context deadline exceeded"

### ISSUE THREAD

**Issue by l-we** [distribworks/dkron#545]
## Expected Behavior
Agent working normally

## Actual Behavior
```
Caught signal: hangupReloading configuration...
time="2019-06-12T19:37:49+08:00" level=info msg="agent: Received event" event=member-update node=file-nok
time="2019-06-12T19:38:00+08:00" level=info msg="agent: Sending query" job_name=mc node=file-nok query="run:job"
time="2019-06-12T19:38:00+08:00" level=info msg="agent: Received event" event="query: run:job" node=file-nok
time="2019-06-12T19:38:00+08:00" level=info msg="agent: Starting job" job=mc node=file-nok
time="2019-06-12T19:38:00+08:00" level=info msg="agent: Sending query" job_name=lau node=file-nok query="run:job"
time="2019-06-12T19:38:05+08:00" level=error msg="grpc: error dialing." err="context deadline exceeded" node=file-nok server_addr="192.168.61.20:0"
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x83cc72]

goroutine 215 [running]:
google.golang.org/grpc.(*ClientConn).Close(0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.18.0/clientconn.go:853 +0x42
panic(0x117ce60, 0x2069ab0)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
google.golang.org/grpc.(*ClientConn).Invoke(0x0, 0x164c360, 0xc000034098, 0x131f0a5, 0x13, 0x1256600, 0xc000732d50, 0x12dd980, 0xc0008002d0, 0x0, ...)
	/go/pkg/mod/google.golang.org/grpc@v1.18.0/call.go:32 +0x38
github.com/victorcoder/dkron/proto.(*dkronClient).GetJob(0xc0001a2480, 0x164c360, 0xc000034098, 0xc000732d50, 0x0, 0x0, 0x0, 0x2083e00, 0xc0004d9b00, 0x3)
	/dkron/proto/dkron.pb.go:439 +0xd2
github.com/victorcoder/dkron/dkron.(*GRPCClient).CallGetJob(0xc0008b2600, 0xc000521900, 0xf, 0xc0005218f0, 0xe, 0x0, 0x0, 0x0)
	/dkron/dkron/grpc.go:272 +0x22c
github.com/victorcoder/dkron/dkron.(*Agent).eventLoop(0xc000194360)
	/dkron/dkron/agent.go:500 +0x14fb
created by github.com/victorcoder/dkron/dkron.(*Agent).Start
	/dkron/dkron/agent.go:107 +0x3b9
```

## Steps to Reproduce the Problem

```
backend: consul
backend-machine: 192.168.16.20:8500
server: true
tags:
  role: file-nok
join:
  - 192.168.61.20
  - 192.168.61.50
  - 192.168.61.90
```

## Specifications

  - Version: 1.2.4
  - Platform: Linux
  - Backend store: consul

**Comment by victorcoder** [comment]
Hi @l-we, does it happens only after reloading configuration?

**Comment by l-we** [comment]
> 嗨@ l-we，只有在重新加载配置后才会发生吗？

yes,

**Comment by fopina** [comment]
not sure if this helps but I use consul with [vault](https://www.vaultproject.io/) and `context deadline exceeded` happens quite often. It's a generic message for consult taking too long to respond and, in my case, it is due to leader re-election in consul cluster...

**Comment by thavlik** [comment]
Similar problem here. Not sure if it's the same one, but I'm running dkron:latest, and the following script...

```
#!/bin/bash
read -r -d '' BODY <<EOM
{
    "name": "job1",
    "schedule": "@every 10s",
    "timezone": "Europe/Berlin",
    "owner": "Platform Team",
    "owner_email": "platform@example.com",
    "disabled": true,
    "tags": {
        "dkron_server": "true"
    },
    "retries": 2,
    "parent_job": "parent_job",
    "processors": {
        "files": {
        "forward": true
        }
    },
    "concurrency": "allow",
    "executor": "shell",
    "executor_config": {
        "command": "echo 'Hello from Dkron'"
    }
}
EOM

curl -X POST \
    http://localhost:8080/v1/jobs \
    -H 'Content-Type: application/json' \
    -d "$BODY"
```

...produces this error:

```
dkron_1         | time="2019-07-09T14:51:55Z" level=info msg="api: Running HTTP server" address=":8080" node=6a3d5db0d3f3
dkron_1         | time="2019-07-09T14:51:55Z" level=info msg="agent: Listen for events" node=6a3d5db0d3f3
dkron_1         | time="2019-07-09T14:51:55Z" level=info msg="dkron: monitoring leadership" node=6a3d5db0d3f3
dkron_1         | time="2019-07-09T14:51:56Z" level=info msg="agent: Received event" event=member-update node=6a3d5db0d3f3
***THE ABOVE BASH SCRIPT WAS ISSUED AT THIS POINT***
dkron_1         | time="2019-07-09T14:52:30Z" level=error msg="grpc: error dialing." error="context deadline exceeded" method=CallSetJob node=6a3d5db0d3f3 server_addr=
```

**Comment by victorcoder** [comment]
@thavlik your case seems a different issue, `dkron:latest` probably points to b2, I don't recommend to pin to latest unless you know what you want.

Probably you get the error because there is no leader elected by default using a single node unless you specify `--bootstrap-expect=1` as param

**Comment by thavlik** [comment]
Roger that! Happy to see b2 under development btw :)

**Comment by vaibhavfareye** [comment]
@victorcoder Any update on how to fix this on v2.0.0-rc9 ?
I am facing this issue on 
AMAZON AMI , rhel fedora

**Comment by victorcoder** [comment]
@vaibhavfareye This issue became a mess, could you open an issue with the relevant logs and describing the problem you are having?
